pytket/extensions/qiskit/qiskit_convert.py

```diff
@@ -325,6 +328,14 @@ def add_qiskit_data(self, data: "QuantumCircuitData") -> None:
                 q_ctrl_box = QControlBox(c_box, instr.num_ctrl_qubits)
                 self.tkc.add_qcontrolbox(q_ctrl_box, qubits)
 
+            elif isinstance(instr, Initialize):
+                # Check that the Initialize object is constructed with a list:
+                # https://qiskit.org/documentation/stubs/qiskit.circuit.QuantumCircuit.initialize.html
+                if isinstance(instr.params, list):
```

Comment:
As it stands this code will give incorrect results when the params are not given as a list, since no gate will be added. We should either handle the other cases, or fail with an error if we encounter them.

I think it wouldn't be too hard to handle them. For a string (such as "1++") it would just be a bunch of 1-qubit gates depending on the string. For an int it would be a bunch of X gates depending on where the 1s are in the binary representation.

Reply:
Thanks, I mistakenly thought that an error would be thrown if `instr.params` was not a list but I now realise that the code would keep running and the resulting circuit would just omit the state preparation gadget making it incorrect. 

I will see if I can handle this in a better way